src/main/java/com/aws/greengrass/componentmanager/ComponentManager.java

```diff
@@ -355,18 +366,21 @@ void prepareArtifacts(ComponentIdentifier componentIdentifier, List<ComponentArt
                 .addKeyValue(PACKAGE_IDENTIFIER, componentIdentifier).log();
 
         for (ComponentArtifact artifact : artifacts) {
-            // check disk space before download
-            // TODO: [P41215447]: Check artifact size for all artifacts to download early to fail early
-            long usableSpaceBytes = componentStore.getUsableSpace();
-            if (usableSpaceBytes < DEFAULT_MIN_DISK_AVAIL_BYTES) {
-                throw new SizeLimitException(
-                        String.format("Disk space critical: %d bytes usable, %d bytes minimum allowed",
-                                usableSpaceBytes, DEFAULT_MIN_DISK_AVAIL_BYTES));
-            }
             ArtifactDownloader downloader = artifactDownloaderFactory
                     .getArtifactDownloader(componentIdentifier, artifact, packageArtifactDirectory);
-
             if (downloader.downloadRequired()) {
+                if (!downloader.downloadReady()) {
```

Comment:
maybe `canDownload`?

Reply:
Have that return an Optional string so that we can tell the user _why_ it is unable to download, not only that it can't do it.